code/modules/status_system/statusFoodBuffs.dm

```diff
@@ -302,7 +302,10 @@
 	proc/cast()
 
 		var/turf/T = get_step(owner,owner.dir)
-		T = get_step(T,owner.dir)
+		var/whileexit = 1
+		while((get_dist(owner,T) < range) && (whileexit < 20))
+			T = get_step(T,owner.dir)
+			whileexit += 1
```

Comment:
```suggestion
		var/range = 1
		while((get_dist(owner,T) < range) && (range < 20))
			T = get_step(T,owner.dir)
			range++
```

Reply:
Range is defined inside of Fire Burp Status Effect code, and is deeply inappropriate to use there.

Reply:
Range is used to make each burp have a different max range, and changing the code that way would make it so that all fireburps are twenty tiles long.

Reply:
Actually, it makes them one tile long, testing shows. Setting default Range to 0 makes it 0 tiles long, and setting default Range to 2 makes it 19 tiles long.

Reply:
`whileexit` is not a great variable name but vode is right that you're breaking everything by naming it `range`.

Reply:
there, renamed from `range`